sp_HumanEvents/sp_HumanEventsBlockViewer.sql

```diff
@@ -624,8 +640,62 @@ ADD query_text AS
    NCHAR(11),N'?'),NCHAR(8),N'?'),NCHAR(7),N'?'),NCHAR(6),N'?'),NCHAR(5),N'?'),NCHAR(4),N'?'),NCHAR(3),N'?'),NCHAR(2),N'?'),NCHAR(1),N'?'),NCHAR(0),N'?')
 PERSISTED;
 
+ALTER TABLE #blocking
+ADD blocking_desc AS 
+    '(' + CAST(blocking_spid AS VARCHAR(max)) + ':' + CAST(blocking_ecid AS VARCHAR(max)) + ')' PERSISTED,
+    blocked_desc AS
+    '(' + CAST(blocked_spid AS VARCHAR(max)) + ':' + CAST(blocked_ecid AS VARCHAR(max)) + ')' PERSISTED;
+    
+
 IF @debug = 1 BEGIN SELECT '#blocking' AS table_name, * FROM #blocking AS wa OPTION(RECOMPILE); END;
 
+WITH hierarchy AS 
+(
+    SELECT 
+        monitor_loop, 
+        blocking_desc,
+        blocked_desc,
+        level = 0,
+        sort_order = blocking_desc + ' <-- ' + blocked_desc
+    FROM #blocking b
+    WHERE NOT EXISTS (
+        SELECT *
+        FROM #blocking b2
+        WHERE b2.monitor_loop = b.monitor_loop
+        AND b2.blocked_desc = b.blocking_desc
+    )
+
+    UNION ALL
+
+    SELECT 
+        bg.monitor_loop,
+        bg.blocking_desc,
+        bg.blocked_desc,
+        h.level + 1,
+        h.sort_order + ' ' + bg.blocking_desc + ' <-- ' + bg.blocked_desc
+    FROM hierarchy h
+    JOIN #blocking bg
+        ON bg.monitor_loop = h.monitor_loop
+        AND bg.blocking_desc = h.blocked_desc
+)
+UPDATE #blocked
+SET blocking_level = h.level,
+    sort_order = h.sort_order
+FROM #blocked b
+JOIN hierarchy h
+    ON h.monitor_loop = b.monitor_loop
+    AND h.blocking_desc = b.blocking_desc
+    AND h.blocked_desc = b.blocked_desc;
+
+UPDATE #blocking
```

Comment:
Update the `#blocking` processes to indicate the blocking tree info (use the corresponding info in `#blocked`